### PR TITLE
[feature] Added support for openwisp_notifications

### LINKED ISSUE #214: [feature] Add openwisp-notifications
- automatically set site domain using the FQDN (by default is example.com and the notification JS widget will send users to example.com if not set correctly) and add suggestion to revise this manually once the installation is completed
  - ensure the [settings for the notifications module are present](https://github.com/openwisp/openwisp-controller/pull/265/files#diff-9dec27df1ed1039b50d2b96520f1de08)

This is required to release openwisp-controller 0.8.

### PR DESCRIPTION
**Pull request by pandafy**
- [x] Added support for openwisp_notifications integration in openwisp_controller
 - [x] Updated docs to ask user to update Site object
 - [x] Added message to ask user to update Site object when role completes execution
 - [x] Updated tests to run integration tests for notifications

Fixes #214 

--------

Blocked by:
 - [x] https://github.com/openwisp/openwisp-controller/pull/316 -  Refactoring of routes and URLs of openwisp_notifications
 - [x] https://github.com/openwisp/openwisp-notifications/pull/143 - Fixed stalling of migrate command when celery broker is unreachable
 - [x] https://github.com/openwisp/openwisp-notifications/pull/152 - Release openwisp-notifications==0.3.0
 - [x] Release openwisp-controller==0.8
------

I have used this playbook for manually testing this patch on Ubuntu 20.04

```
- hosts: openwisp2
  become: "{{ become | default('yes') }}"
  roles:
    - openwisp.openwisp2
  vars:
    openwisp2_controller_pip: https://github.com/pandafy/openwisp-controller/tarball/fix-notification-settings
    openwisp2_extra_python_packages:
      - django-flat-json-widget
      - https://github.com/openwisp/openwisp-notifications/tarball/master
    openwisp2_default_from_email: "openwisp2@openwisp2.mydomain.com"
```

### CODE REVIEW

File: defaults/main.yml [hunk 1]
```diff
@@ -52,6 +53,7 @@ openwisp2_sentry:
 openwisp2_default_cert_validity: 1825
 openwisp2_default_ca_validity: 3650
 openwisp2_default_organization_id: 78f74fdd-67c6-4064-97e1-ce761da30745
+openwisp2_notifications_delete_old_notifications: 90
```
Comment (nemesisdesign) [inline defaults/main.yml:56]: don't forget to add these two new variables in the [Role Variables docs](https://github.com/openwisp/ansible-openwisp2#role-variables)

**Review by nemesisdesign — COMMENTED** [review]
Looks good, although the build is failing for idempotence test, I remember you fixed that, is this branch already up to date @pandafy?

**Comment by pandafy** [comment]
@nemesisdesign #219 skipped idempotence test for "Install custom OpenWISP 2 Python packages" task which installs packages from variables defined for openwisp modules. In this patch I have used `openwisp2_extra_python_packages` variable to install openwisp-notification due to `--no-deps` flag set in `Install custom OpenWISP 2 Python packages` task.(See #193).

I will update it once new version of openwisp-controller and openwisp-notifications is released.

**Review by nemesisdesign — COMMENTED** [review]
We'll also need to merge this one https://github.com/openwisp/ansible-openwisp2/pull/194 before or after this.

File: molecule/resources/converge.yml [hunk 2]
```diff
@@ -6,6 +6,11 @@
 
   vars:
     openwisp2_network_topology: true
+    # Remove when openwisp-controller==0.8 is released
+    openwisp2_controller_pip: https://github.com/openwisp/openwisp-controller/tarball/master
+    openwisp2_extra_python_packages:
+      - django-flat-json-widget
+      - https://github.com/openwisp/openwisp-notifications/tarball/master
```
Comment (nemesisdesign) [inline molecule/resources/converge.yml:13]: This shouldn't be needed anymore, we should set ow-controller to 0.8 here: https://github.com/openwisp/ansible-openwisp2/blob/master/defaults/main.yml#L4 and network-topology to 0.5

**Review by nemesisdesign — CHANGES_REQUESTED** [review]

**Review by nemesisdesign — APPROVED** [review]
Great work @pandafy :+1:

File: README.md [hunk 3]
```diff
@@ -536,6 +537,8 @@ Below are listed all the variables you can customize (you may also want to take
     openwisp2_controller_pip: false
     openwisp2_users_pip: false
     openwisp2_utils_pip: false
+    openwisp2_notifications_pip: false
+    openwisp2_django_netjsonconfig_pip: false
```
Comment (nemesisdesign) [inline README.md:541]: mm.. this should have been removed

**Review by nemesisdesign — CHANGES_REQUESTED** [review]
Wait a minute, I spotted one thing to correct.

**Review by nemesisdesign — APPROVED** [review]
Thanks @pandafy :+1: :tada: :+1:

**Pull request merged**